src/transformers/models/siglip/modeling_siglip.py

```diff
@@ -926,8 +926,11 @@ def __init__(self, config: SiglipTextConfig):
         self.embeddings = SiglipTextEmbeddings(config)
         self.encoder = SiglipEncoder(config)
         self.final_layer_norm = nn.LayerNorm(embed_dim, eps=config.layer_norm_eps)
-
-        self.head = nn.Linear(embed_dim, embed_dim)
+        self.head = (
+            nn.Linear(embed_dim, embed_dim, bias=False)
+            if config.has_head
+            else lambda x: x @ torch.eye(embed_dim, device=x.device)
+        )
```

Comment:
No 😅 the idea is that you set the self.head's weights to `eye` in the conversion script. This way it's a no-op!